govcd/cse_yaml.go

```diff
@@ -0,0 +1,410 @@
+package govcd
+
+import (
+	"fmt"
+	semver "github.com/hashicorp/go-version"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"sigs.k8s.io/yaml"
+	"strings"
+)
+
+// updateCapiYaml takes a YAML and modifies its Kubernetes Template OVA, its Control plane, its Worker pools
+// and its Node Health Check capabilities, by using the new values provided as input.
+// If some of the values of the input is not provided, it doesn't change them.
+// If none of the values is provided, it just returns the same untouched YAML.
+func (cluster *CseKubernetesCluster) updateCapiYaml(input CseClusterUpdateInput) (string, error) {
+	if cluster == nil {
+		return cluster.capvcdType.Spec.CapiYaml, fmt.Errorf("receiver cluster is nil")
```

Comment:
Not me, but linter complains - any risk of `capvcdType` being `nil`?

Reply:
Good catch, indeed it would cause a really dumb panic